lib/smart_proxy_remote_execution_ssh/runners/script_runner.rb

```diff
@@ -258,32 +257,31 @@ def indent_multiline(string)
     end
 
     def should_cleanup?
-      @session && !@session.closed? && @cleanup_working_dirs
+      @session && @cleanup_working_dirs
     end
 
-    def session
-      @session ||= begin
-                     @logger.debug("opening session to #{@ssh_user}@#{@host}")
-                     Net::SSH.start(@host, @ssh_user, ssh_options)
-                   end
+    def session(command)
+      @session = true
+      command_out, slave = PTY.open
+      command_in, write = IO.pipe
+      command_pid = spawn(({'SSHPASS' => @ssh_password} if @ssh_password), '/usr/bin/sshpass', '-e', '/usr/bin/ssh', @host, *ssh_options, command, :in => command_in, :out => slave)
```

Comment:
Thinking about this, we still need a pty, but we need it on the remote side. Could you try experimenting if we could do without dealing with ptys here and just pass `-t` to `ssh`?